src/auth/envDetect.ts

```diff
@@ -36,10 +36,12 @@ export async function getEnv() {
       env = GCPEnv.APP_ENGINE;
     } else if (isCloudFunction()) {
       env = GCPEnv.CLOUD_FUNCTIONS;
-    } else if (await isKubernetesEngine()) {
-      env = GCPEnv.KUBERNETES_ENGINE;
     } else if (await isComputeEngine()) {
-      env = GCPEnv.COMPUTE_ENGINE;
+      if (await isKubernetesEngine()) {
```

Comment:
we end up performing one more check for kubernetes, but we use the fast path `gcpMetadata.isAvailable` rather than the slow `gcpMetadata.instance('attributes/cluster-name')` for the initial check, correcting issues with `@nodejs/logging` performance.